grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/DefaultGrpcExceptionHandlerFunction.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.common.grpc;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.nio.channels.ClosedChannelException;
+
+import com.google.common.base.Strings;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import com.linecorp.armeria.client.UnprocessedRequestException;
+import com.linecorp.armeria.client.circuitbreaker.FailFastException;
+import com.linecorp.armeria.client.grpc.GrpcClientBuilder;
+import com.linecorp.armeria.common.ClosedSessionException;
+import com.linecorp.armeria.common.ContentTooLargeException;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.TimeoutException;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.grpc.GrpcExceptionHandlerFunction;
+import com.linecorp.armeria.common.grpc.GrpcStatusFunction;
+import com.linecorp.armeria.common.grpc.StackTraceElementProto;
+import com.linecorp.armeria.common.grpc.StatusCauseException;
+import com.linecorp.armeria.common.grpc.ThrowableProto;
+import com.linecorp.armeria.common.grpc.protocol.ArmeriaStatusException;
+import com.linecorp.armeria.common.stream.ClosedStreamException;
+import com.linecorp.armeria.common.util.Exceptions;
+import com.linecorp.armeria.server.RequestTimeoutException;
+import com.linecorp.armeria.server.grpc.GrpcServiceBuilder;
+
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.grpc.Status.Code;
+import io.netty.handler.codec.http2.Http2Error;
+import io.netty.handler.codec.http2.Http2Exception;
+
+public final class DefaultGrpcExceptionHandlerFunction implements GrpcExceptionHandlerFunction {
```

Comment:
Should we move this class to `com.linecorp.armeria.common.grpc` and make it package-private?
```suggestion
final class DefaultGrpcExceptionHandlerFunction implements GrpcExceptionHandlerFunction {
```

Reply:
Applied feedback. Thanks